### ISSUE TITLE
network weave-net not eligible for docker service

### ISSUE THREAD

**Issue by ackris** [weaveworks/weave#3074]
Hi Weave Team :)

I am facing this peculiar issue of weave-net not working with docker services. The main purpose of using weave-net is to leverage the multi-cast utility of this network. For further help, here is my docker info:

```
-bash-4.2$ docker info
Containers: 22
Running: 13
Paused: 0
Stopped: 9
Images: 97
Server Version: 17.03.2-ee-4
Storage Driver: devicemapper
 Pool Name: docker-253:3-203091234-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 19.63 GB
 Data Space Total: 107.4 GB
 Data Space Available: 11.94 GB
 Metadata Space Used: 24.54 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.123 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay weaveworks/net-plugin:latest
Swarm: active
 NodeID: l4zplq8ckl7e53fxms3mxe9q3
 Is Manager: true
 ClusterID: 3rphbfgcq4oalh9gyuvfcmidd
 Managers: 1
 Nodes: 2
Orchestration:
 Task History Retention Limit: 5
Raft:
 Snapshot Interval: 10000
 Number of Old Snapshots to Retain: 0
 Heartbeat Tick: 1
 Election Tick: 3
Dispatcher:
 Heartbeat Period: 5 seconds
 CA Configuration:
 Expiry Duration: 3 months
External CAs:
 cfssl: https://10.151.1.205:12381/api/v1/cfssl/sign
Node Address: 10.151.1.205
Manager Addresses:
 10.151.1.205:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
 Profile: default
Kernel Version: 3.10.0-514.21.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 19.44 GiB
Name: ucp-node
ID: YJ2D:L5XH:Z7VB:2Z5X:CSMY:SC3D:WAG6:YYYD:MITU:CMAO:ZZ4Q:YO5P
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
```

Weave-net plugin installation method:
```
docker plugin install weaveworks/net-plugin
docker network create --driver weaveworks/net-plugin:latest --attachable weave-net 
```

Error:
**## network weave-net not eligible for docker service** 

The output of docker network inspect weave-net
```
docker network inspect weave-net 
[
 {
        "Name": "weave-net",
        "Id": "iwiarv4plpnwjde78dluxt46j",
        "Created": "0001-01-01T00:00:00Z",
        "Scope": "swarm",
        "Driver": "weaveworks/net-plugin:latest",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": []
        },
        "Internal": false,
        "Attachable": true,
        "Containers": null,
        "Options": null,
        "Labels": null
    }
]
```
I have also tried with  `docker network create --scope global --driver  weaveworks/net-plugin:latest --attachable  weave-net` but I am getting error like `unknown flag --scope`.

The output of my ps aux | grep weaver
```
ps aux | grep weaver

crestel  20230  0.0  0.0 112648   960 pts/1    R+   14:56   0:00 grep --color=auto weaver

root     25983  0.0  0.2 865020 58088 ?        Ssl  Jul19   4:44 /home/weave/weaver --port=6783 --datapath=datapath --host-root=/host --proc-path=/host/proc --http-addr=127.0.0.1:6784 --status-addr=0.0.0.0:6782 --no-dns --ipalloc-range=10.32.0.0/12 --nickname ucp-node --log-level=debug --db-prefix=/host/var/lib/weave/weave --plugin-v2 --plugin-mesh-socket= --docker-api= 10.151.1.205
```
I have verified that the plugin is enabled. 

```
-bash-4.2$ docker plugin ls

ID                  NAME                           DESCRIPTION                   ENABLED
a3ee6765207c        weaveworks/net-plugin:latest   Weave Net plugin for Docker   true
```
I have gone through  [#Issue 2727](https://github.com/weaveworks/weave/pull/2727). But it is not working out. Where am i making a mistake?

Regards and Thanks in advance :)

**Comment by brb** [comment]
Thanks for the report.

Have you tried creating the network without `--attachable`? It should make it to work with Docker services.

**Comment by ackris** [comment]
@brb 

Thanks for the reply. I haven't tried it yet. I will work with your suggestion and come back to you :)

**Comment by ackris** [comment]
Hi @brb 

I am getting the same error. 

1) Created the network sans --attachable flag
```
docker network create --driver weaveworks/net-plugin:latest  weave-network 
34krecjt6xdqhzyyuw8lrq7rb
```

2.Output of Docker inspect
```
-bash-4.2$ docker inspect weave-network 
[
    {
        "Name": "weave-network",
        "Id": "34krecjt6xdqhzyyuw8lrq7rb",
        "Created": "0001-01-01T00:00:00Z",
        "Scope": "swarm",
        "Driver": "weaveworks/net-plugin:latest",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": []
        },
        "Internal": false,
        "Attachable": false,
        "Containers": null,
        "Options": null,
        "Labels": null
    }
]
```

3) Tried to create service from docker UCP UI and getting same error like `network weave-network is not eligible for docker services.
`
Where am i making a mistake?

Regards
Aditya

**Comment by ackris** [comment]
@brb @rade 

Any solution to this issue?

Regards

**Comment by brb** [comment]
@ackris are you able to start service manually via `docker service create --network weave-network --replicas=2 nginx`?

**Comment by ackris** [comment]
Hi @brb @rade 

Please have a look at this [link](https://github.com/docker/compose/issues/5065). I am facing this new issue. 

I have successfully created a weave-net network. But when I have given docker-stack deploy, it is crashing (sort of!). I don't know if it has something to do with weave-net. Please verify whether I have mentioned weave-net correctly in this docker-compose file. 

Regards
Aditya

**Comment by brb** [comment]
@ackris The issue you linked has been resolved, right? Can we close this #3074 issue?

**Comment by ackris** [comment]
Hi @brb 
I am closing this issue, and thanks for the support. 

I will come back if this error persists :)

Have a good day and regards!
